src/diffpy/utils/parsers/loadmetafile.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+##############################################################################
+#
+# diffpy.utils      by DANSE Diffraction group
+#                   Simon J. L. Billinge
+#                   (c) 2010 The Trustees of Columbia University
+#                   in the City of New York.  All rights reserved.
+#
+# File coded by:
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE_DANSE.txt for license information.
+#
+##############################################################################
+
+import pathlib
+import json
+
+# FIXME: add support for yaml, xml
+supported_formats = ['.json']
+
+
+def load_PDF_into_db(dbname, pdfname, hddata: dict, rv: list, show_path=True):
+    """Load an entry consisting of PDF header and base data into a database file.
+
+    Requires hdata and rv generated from loadData.
+
+    dbname          -- name of the database file to load an entry into.
+    pdfname         -- name of the PDF file.
```

Comment:
why are you limiting it to a pdf file?  couldn't it be any file that has the right header/column data structure?  iq file? chi file? sq file? etc.  Make it as general as possible so it is more reusable.  Basically, any file that loadData can read....

Reply:
No longer has limit, but has an extra dt_colnames input requirement since before we were assuming column names as 'r' and 'gr'.